tests/test_class_sh_shared_ptr_copy_move.cpp

```diff
@@ -0,0 +1,71 @@
+#include "pybind11_tests.h"
+
+#include <pybind11/smart_holder.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace pybind11_tests {
+namespace {
+
+const std::string fooNames[] = {"ShPtr_", "SmHld_"};
+
+template <int SerNo>
+struct Foo {
+    std::string hisotry;
+    Foo(const std::string &hisotry_) : hisotry(hisotry_) {}
+    Foo(const Foo &other) = delete;
+    Foo(Foo &&other)      = delete;
```

Comment:
The purpose of the `history` member is completely lost: the move and copy assignment operators are not `delete`d, but also not tracked. How those behave is exactly what we want to test here. (At the moment idk what our conclusion will be when we see the results on all platforms, but it's important that we know what they are.)

Reply:
Indeed, there should now be tracking of the special member functions.